### PR TITLE
add docs for special buttons

### PR DESCRIPTION
**Pull request by robin-drexler**
### Background
See https://github.com/Shopify/customer-account-web/pull/3362 for context and 🎩  instructions

### Checklist

- [x] I have :tophat:'d these changes
- [ ] I have updated relevant documentation

### CODE REVIEW

File: packages/ui-extensions/src/surfaces/customer-account/api/docs.ts [hunk 1]
```diff
@@ -77,3 +78,48 @@ export interface Docs_Standard_QueryApi
...
+  > {}
+export interface Docs_Page_Button_SecondaryAction
+  extends Pick<
+    ButtonProps,
+    'onPress' | 'to' | 'loading' | 'loadingLabel' | 'disabled'
+  > {
+  /**
+   * A label used for buyers using assistive technologies. Needed because `children` passed to this component will be discarded.
+   */
+  accessibilityLabel: ButtonProps['accessibilityLabel'];
```
Comment (robin-drexler) [inline packages/ui-extensions/src/surfaces/customer-account/api/docs.ts:100]: One downside is that this messes with the order of props in the doc. the one you override always come last.

We could do something like:

```ts
export interface Docs_Page_Button_SecondaryAction {
  to: ButtonProps['to']
  onPress: ButtonProps['onPress']
  // etc..
}
```

But then we'd need to copy paste the comment for each prop from the original component over and over again.

File: packages/ui-extensions/src/surfaces/customer-account/api/docs.ts [hunk 2]
```diff
@@ -77,3 +78,48 @@ export interface Docs_Standard_QueryApi
...
+    | 'loading'
+    | 'loadingLabel'
+    | 'disabled'
+    | 'accessibilityLabel'
+  > {}
+
+export interface Docs_OrderActionMenu_Button
+  extends Pick<
+    ButtonProps,
+    'onPress' | 'loading' | 'loadingLabel' | 'disabled' | 'accessibilityLabel'
```
Comment (brianshen1990) [inline packages/ui-extensions/src/surfaces/customer-account/api/docs.ts:117]: Do we accept all of this? I think we only picked the
`text` `to` 
https://github.com/Shopify/customer-account-web/blob/main/app/foundation/extensibility/extensions/menu-items/useMenuItems.tsx#L59-L67

Comment (robin-drexler) [reply packages/ui-extensions/src/surfaces/customer-account/api/docs.ts:117]: I think we will, but I could remove it until we actually do and we'd add it once it's there for real.

File: packages/ui-extensions/src/surfaces/customer-account/api/docs.ts (hunk 1 above)
Comment (robin-drexler) [reply packages/ui-extensions/src/surfaces/customer-account/api/docs.ts:100]: I'll have a look at adding this to the host so we don't need to bump this package over and over again.

File: packages/ui-extensions/src/surfaces/customer-account/api/docs.ts (hunk 2 above)
Comment (robin-drexler) [reply packages/ui-extensions/src/surfaces/customer-account/api/docs.ts:117]: I'll check how much effort it is to add it to the host. I'm ok, if this PR here is open for a day or so

Comment (robin-drexler) [reply packages/ui-extensions/src/surfaces/customer-account/api/docs.ts:118]: It's not a lie anymore :D

File: packages/ui-extensions/src/surfaces/customer-account/api/docs.ts [hunk 3]
```diff
@@ -77,3 +78,49 @@ export interface Docs_Standard_QueryApi
...
+    | 'to'
+    | 'loading'
+    | 'loadingLabel'
+    | 'disabled'
+    | 'accessibilityLabel'
+  > {}
+export interface Docs_Page_Button_SecondaryAction
+  extends Pick<
+    ButtonProps,
+    'onPress' | 'to' | 'loading' | 'loadingLabel' | 'disabled'
```
Comment (lsit) [inline packages/ui-extensions/src/surfaces/customer-account/api/docs.ts:95]: I don't think we're adding `loading`, `loadingLabel`, `disabled` to the secondaryAction (i.e. back button).

**Review by kirakik — APPROVED** [review]

File: packages/ui-extensions/src/surfaces/customer-account/api/docs.ts (hunk 3 above)
Comment (robin-drexler) [reply packages/ui-extensions/src/surfaces/customer-account/api/docs.ts]: good catch! fixed.

**Review by lsit — APPROVED** [review]

**Pull request merged**